execution/executor/src/lib.rs

```diff
@@ -964,18 +964,9 @@ impl<V: VMExecutor> BlockExecutor for Executor<V> {
             )?;
         }
 
-        let mut write_lock = self.cache.write();
-        let arc_blocks = block_ids
-            .iter()
-            .map(|id| write_lock.get_block(id))
-            .collect::<Result<Vec<_>, Error>>()?;
-        let blocks = arc_blocks.iter().map(|b| b.lock()).collect::<Vec<_>>();
-
-        for block in blocks {
-            block.output().executed_trees().state_tree().prune()
-        }
-
-        write_lock.prune(ledger_info_with_sigs.ledger_info())?;
+        self.cache
+            .write()
+            .prune(ledger_info_with_sigs.ledger_info())?;
```

Comment:
So, what's the rate of updating the block_tree_root to updating it on commit?  Seems fine to me, as long as it isn't going to all of a sudden explode in size.